pytensor/link/numba/dispatch/slinalg.py

```diff
@@ -267,14 +292,79 @@ def solve_triangular(a, b):
         res = _solve_triangular(a, b, trans, lower, unit_diagonal)
         if check_finite:
             if np.any(np.bitwise_or(np.isinf(res), np.isnan(res))):
-                raise ValueError(
+                raise np.linalg.LinAlgError(
                     "Non-numeric values (nan or inf) returned by solve_triangular"
                 )
         return res
 
     return solve_triangular
 
 
+def _cholesky(a, lower=False, overwrite_a=False, check_finite=True):
+    return linalg.cholesky(
+        a, lower=lower, overwrite_a=overwrite_a, check_finite=check_finite
+    )
+
+
+@overload(_cholesky)
+def cholesky_impl(A, lower=0, overwrite_a=False, check_finite=True):
+    ensure_lapack()
+    _check_scipy_linalg_matrix(A, "cholesky")
+    dtype = A.dtype
+    w_type = _get_underlying_float(dtype)
+    numba_potrf = _LAPACK().numba_xpotrf(dtype)
+
+    def impl(A, lower=0, overwrite_a=False, check_finite=True):
+        _N = np.int32(A.shape[-1])
+        if A.shape[-2] != _N:
+            raise linalg.LinAlgError("Last 2 dimensions of A must be square")
+
+        if check_finite:
+            _check_finite_matrix(A, "cholesky")
+
+        UPLO = val_to_int_ptr(ord("L") if lower else ord("U"))
+        N = val_to_int_ptr(_N)
+        LDA = val_to_int_ptr(_N)
+        INFO = val_to_int_ptr(0)
+
+        if not overwrite_a:
+            A_copy = _copy_to_fortran_order(A)
+        else:
+            A_copy = A
+
+        numba_potrf(
+            UPLO,
+            N,
+            A_copy.view(w_type).ctypes,
+            LDA,
+            INFO,
+        )
+
+        return A_copy
+
+    return impl
+
+
+@numba_funcify.register(Cholesky)
+def numba_funcify_Cholesky(op, node, **kwargs):
+    lower = op.lower
+    overwrite_a = False
+    check_finite = op.on_error == "raise"
```

Comment:
I think this flag was created because Scipy Cholesky always raised, and it was needed to catch it for some use cases. Like even if used in a switch statement where we would ignore the values it would crash the program.

That to say that raise was not necessarily a feature, but the default behavior. The flag feature was to allow suppressing the warning. So we may not need to add the extra flag logic here?

I could be thinking about the flag of another Op, but git history should confirm.

Reply:
Well there is a `check_finite` flag on the original scipy function. I was going to point out that the signatures didn't match and ask if it was appropriate to change it in this PR, but nothing gets past you.

Reply:
What does the final suggestion look like?

Reply:
Change the signature on the Cholesky `Op` to use `check_finite` instead of `on_error`, which would take a bool instead of a string. But that would break existing code, so I'd have to add a deprecation warning too.

Reply:
I don't think check finite has the same goal. That's just a check the input has valid values.

The op error thing was about not raising an error when the decomposition fails. I assume this can happen even with finite values?

Reply:
You can add the check_finite kwarg though.

About the raise, what would happen now if the matrix is not SPD?

Reply:
Oh I didn't think about the difference between input and output checking. The way I'm doing the flag is currently wrong -- I'm checking the output, not the input. 

Right now if the matrix isn't PSD it will return the input matrix without changes, which is clearly wrong. I'll fix that. So the `on_error` flag should return a matrix of all nan instead of raising a "not PSD" error?